File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();

        int minprice = INT_MAX;
        int maxprice = 0;

        int current;

        for(int i=0; i<n; i++) {
            current = prices[i];
            if(current < minprice) {
                minprice = current; 
            }
            else {
                int profit = current - minprice;
                if(profit > maxprice) {
                    maxprice = profit;
                }
            }
        }
        return maxprice;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
import sys

class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """

        if not prices :
            return 0

        n = len(prices)
        minprice = sys.maxsize
        maxprice = 0

        for currentprice in prices:
            if(currentprice < minprice) :
                minprice = currentprice
            else:
                profit = currentprice - minprice
                if(profit > maxprice) :
                    maxprice = profit
        return maxprice
        